pkg/virt-controller/watch/vm.go

```diff
@@ -231,6 +284,77 @@ func (c *VMController) orphan(cm *controller.VirtualMachineControllerRefManager,
 	return nil
 }
 
+func createDataVolumeManifest(dataVolume *cdiv1.DataVolume, vm *virtv1.VirtualMachine) *cdiv1.DataVolume {
+
+	newDataVolume := dataVolume.DeepCopy()
+
+	labels := map[string]string{}
+	annotations := map[string]string{}
+
+	labels[virtv1.CreatedByLabel] = string(vm.UID)
+	annotations[virtv1.OwnedByAnnotation] = "virt-controller"
+
+	for k, v := range dataVolume.Labels {
+		annotations[k] = v
+	}
+	for k, v := range dataVolume.Labels {
+		labels[k] = v
+	}
+	newDataVolume.ObjectMeta.Labels = labels
+	newDataVolume.ObjectMeta.Annotations = annotations
+
+	tr := true
+	newDataVolume.ObjectMeta.OwnerReferences = []v1.OwnerReference{{
```

Comment:
Can you check if the VM is deleted with an `orphanDelete`, if the VMI deletion is blocked on k8s and openshift `x.10`? I think that https://github.com/kubernetes/kubernetes/issues/56348 is only resolved for `x.11` and newer. `replace` workflows with `kubectl` will be blocked if that is the case. There is an `orphan` method in the controller which would need to be extended with explicit data volume orphan logic.

Reply:
I also think that we will need to adopt the volumes again, but maybe I don't properly get on how they work. Are they wrapping a PVC?

Reply:
yeah, this ownership needs to work just like the VMI. I'll address this.

Reply:
fixed